controller_manager:
  ros__parameters:
    update_rate: 100  # Hz

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    right_arm_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    left_arm_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    right_gripper_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    left_gripper_controller:
      type: joint_trajectory_controller/JointTrajectoryController

joint_state_broadcaster:
  ros__parameters: {}

right_arm_controller:
  ros__parameters:
    joints:
      - RARM_JOINT0
      - RARM_JOINT1
      - RARM_JOINT2
      - RARM_JOINT3
      - RARM_JOINT4
      - RARM_JOINT5
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    state_publish_rate: 50.0
    action_monitor_rate: 20.0
    allow_partial_joints_goal: false
    constraints:
      stopped_velocity_tolerance: 0.01
      goal_time: 0.0

left_arm_controller:
  ros__parameters:
    joints:
      - LARM_JOINT0
      - LARM_JOINT1
      - LARM_JOINT2
      - LARM_JOINT3
      - LARM_JOINT4
      - LARM_JOINT5
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    state_publish_rate: 50.0
    action_monitor_rate: 20.0
    allow_partial_joints_goal: false
    constraints:
      stopped_velocity_tolerance: 0.01
      goal_time: 0.0

right_gripper_controller:
  ros__parameters:
    joints:
      - right_left_finger_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    state_publish_rate: 50.0
    action_monitor_rate: 20.0

left_gripper_controller:
  ros__parameters:
    joints:
      - left_left_finger_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    state_publish_rate: 50.0
    action_monitor_rate: 20.0